pkg/proxy/healthcheck/proxier_health.go

```diff
@@ -44,179 +33,64 @@ type ProxierHealthUpdater interface {
 var zeroTime = time.Time{}
 
-// proxierHealthServer returns 200 "OK" by default. It verifies that the delay between
+// proxierHealthManager returns 200 "OK" by default. It verifies that the delay between
 // QueuedUpdate() calls and Updated() calls never exceeds healthTimeout.
-type proxierHealthServer struct {
-	listener    listener
-	httpFactory httpServerFactory
-	clock       clock.Clock
-
-	addr          string
+type proxierHealthManager struct {
+	clock         clock.Clock
 	healthTimeout time.Duration
 
 	lastUpdated         atomic.Value
 	oldestPendingQueued atomic.Value
-	nodeEligible        atomic.Bool
-}
-
-// NewProxierHealthServer returns a proxier health http server.
-func NewProxierHealthServer(addr string, healthTimeout time.Duration) ProxierHealthUpdater {
-	return newProxierHealthServer(stdNetListener{}, stdHTTPServerFactory{}, clock.RealClock{}, addr, healthTimeout)
 }
 
-func newProxierHealthServer(listener listener, httpServerFactory httpServerFactory, c clock.Clock, addr string, healthTimeout time.Duration) *proxierHealthServer {
-	hs := &proxierHealthServer{
-		listener:      listener,
-		httpFactory:   httpServerFactory,
+// newProxierHealthManager returns a proxier health manager.
+func newProxierHealthManager(c clock.Clock, healthTimeout time.Duration) *proxierHealthManager {
+	return &proxierHealthManager{
 		clock:         c,
-		addr:          addr,
 		healthTimeout: healthTimeout,
 	}
-	// The node is eligible (and thus the proxy healthy) while it's starting up
-	// and until we've processed the first node event that indicates the
-	// contrary.
-	hs.nodeEligible.Store(true)
-	return hs
 }
 
 // Updated indicates that kube-proxy has successfully updated its backend, so it should
 // be considered healthy now.
-func (hs *proxierHealthServer) Updated() {
-	hs.oldestPendingQueued.Store(zeroTime)
-	hs.lastUpdated.Store(hs.clock.Now())
+func (hm *proxierHealthManager) Updated() {
+	hm.oldestPendingQueued.Store(zeroTime)
+	hm.lastUpdated.Store(hm.clock.Now())
 }
 
 // QueuedUpdate indicates that the proxy has received changes from the apiserver but
 // has not yet pushed them to its backend. If the proxy does not call Updated within the
 // healthTimeout time then it will be considered unhealthy.
-func (hs *proxierHealthServer) QueuedUpdate() {
+func (hm *proxierHealthManager) QueuedUpdate() {
 	// Set oldestPendingQueued only if it's currently zero
-	hs.oldestPendingQueued.CompareAndSwap(zeroTime, hs.clock.Now())
+	hm.oldestPendingQueued.CompareAndSwap(zeroTime, hm.clock.Now())
 }
 
-// IsHealthy returns only the proxier's health state, following the same
-// definition the HTTP server defines, but ignoring the state of the Node.
-func (hs *proxierHealthServer) IsHealthy() bool {
-	isHealthy, _, _ := hs.isHealthy()
-	return isHealthy
-}
-
-func (hs *proxierHealthServer) isHealthy() (bool, time.Time, time.Time) {
+func (hm *proxierHealthManager) IsHealthy() (bool, time.Time, time.Time) {
```

Comment:
This method is only used within `pkg/proxy/healthcheck`, so it doesn't need to be exported (likewise in `proxy_health.go`).

(I'm not sure why there was previously also a public `IsHealthy` method wrapping it... check `git blame` and read the PR it came from?)

Reply:
updated!

`IsHealthy` was introduced in https://github.com/kubernetes/kubernetes/pull/111661 as part of `proxierHealthChecker`.
`proxierHealthChecker` is used to limit the API of `ProxierHealthUpdater` for `ServiceHealthServer` consumption.

I guess `IsHealthy` was added to avoid collision because there was already a private method with same name - `isHealthy`